src/embind/embind.js

```diff
@@ -198,21 +198,11 @@ var LibraryEmbind = {
   $createNamedFunction__deps: ['$makeLegalFunctionName'],
   $createNamedFunction: function(name, body) {
     name = makeLegalFunctionName(name);
-#if DYNAMIC_EXECUTION == 0
-    return function() {
-      "use strict";
-      return body.apply(this, arguments);
-    };
-#else
-    /*jshint evil:true*/
-    return new Function(
-        "body",
-        "return function " + name + "() {\n" +
-        "    \"use strict\";" +
-        "    return body.apply(this, arguments);\n" +
-        "};\n"
-    )(body);
-#endif
+    return {
+      [name]: function () {
```

Comment:
Maybe add a comment here? I guess this is using some clever JS VM behavior of guessing the function name based on the property it is written to..?

Reply:
Yes, its a pretty clever solution.  We have tests that verify that it does indeed work (as your can see from the test expectations update).   Added a comment.